### DISCUSSION TITLE
Sample Opaque on Pixel shader with transparency?

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#210] [Feedback & Feature Requests]
One of the drawbacks of janky Eevee refraction shaders is that it swallows all transparent objects behind it. Being able to sample an opaque layer behind an object is nice but would it be possible to also render transparency. That would give Malt a tactical advantage over Eevee I think. Im guessing its not trivial to implement but would it be possible?
![image](https://user-images.githubusercontent.com/88381426/147363893-2b2c43d6-f969-4048-a878-9fbabdf94bc1.png)

**Comment by pragma37** [comment]
At the moment transparent layers are rendered in front to back order, so it's not possible.
It would be trivial to change the draw order, but then only the last n transparent layers would be rendered, instead of the first ones, which would be very problematic on scenes with many overlapped transparent meshes.

I have some changes planned to the way layers are rendered in 1.0, however, so it might be possible to get this working as intended.
